pennylane/circuit_drawer/circuit_drawer.py

```diff
@@ -0,0 +1,322 @@
+# Copyright 2019-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This module contains the CircuitDrawer class which is used to draw CircuitGraph instances.
+"""
+from collections import OrderedDict
+
+import pennylane as qml
+
+from .charsets import UnicodeCharSet
+from .representation_resolver import RepresentationResolver
+from .grid import Grid
+
+# pylint: disable=too-many-branches,too-many-arguments,too-many-return-statements,too-many-statements,consider-using-enumerate,too-many-instance-attributes
+
+
+def _remove_duplicates(input_list):
+    """Remove duplicate entries from a list.
+
+    This operation preserves the order of the list's elements.
+
+    Args:
+        input_list (list[Hashable]): The list whose duplicate entries shall be removed
+
+    Returns:
+        list[Hashable]: The input list without duplicate entries
+    """
+    return list(OrderedDict.fromkeys(input_list))
+
+
+class CircuitDrawer:
+    """Creates a circuit diagram from the operators of a CircuitGraph in grid form.
+
+    Args:
+        raw_operation_grid (list[list[~.Operation]]): The CircuitGraph's operations
+        raw_observable_grid (list[list[qml.operation.Observable]]): The CircuitGraph's observables
+        charset (pennylane.circuit_drawer.CharSet, optional): The CharSet that shall be used for drawing.
+        show_variable_names (bool, optional): Show variable names instead of variable values.
+    """
+
+    def __init__(
+        self,
+        raw_operation_grid,
+        raw_observable_grid,
+        charset=UnicodeCharSet,
+        show_variable_names=False,
+    ):
+        self.charset = charset
+        self.show_variable_names = show_variable_names
+        self.representation_resolver = RepresentationResolver(charset, show_variable_names)
+        self.operation_grid = Grid(raw_operation_grid)
+        self.observable_grid = Grid(raw_observable_grid)
+        self.operation_representation_grid = Grid()
+        self.observable_representation_grid = Grid()
+        self.operation_decoration_indices = []
+        self.observable_decoration_indices = []
```

Comment:
(Very minor thing here: for me, it would be easier to read the initializations in the order of how the arguments were passed and then have a ``newline`` followed. This way it would be easy to follow how the attributes are created.)

Reply:
:+1: